DataMigration/src/main/java/org/sagebionetworks/dian/datamigration/SecureTokenGenerator.java

```diff
@@ -0,0 +1,137 @@
+    // Bridge password must contain at least one symbol ( !"#$%&'()*+,-./:;<=>?@[\]^_`{|}~ );
+    // This subset of symbols was chosen because they would be easier to communicate over the phone
+    private static final String SYMBOL = "!#$%&'*+,-.:;=?@_";
+
+    private static final String ALPHANUMERIC = UPPERCASE_ALPHA + LOWERCASE_ALPHA + NUMERIC;
+    // ALPHANUMERIC was added 3 times to decrease the number of symbols in a password
+    private static final String PASSWORD = ALPHANUMERIC + ALPHANUMERIC + ALPHANUMERIC + SYMBOL;
+
+    public static final SecureTokenGenerator INSTANCE = new SecureTokenGenerator();
+
+    public static final SecureTokenGenerator ARC_ID_INSTANCE =
+            new SecureTokenGenerator(6, new SecureRandom(), NUMERIC);
+
+    /**
+     * I used this website https://asecuritysite.com/encryption/passes and using our parameters,
+     * it said a 9 character password has 66,540,410,775,079,424 available passwords,
+     * and it would take 2108.59 years to crack if requests were sent.
+     */
+    public static final SecureTokenGenerator BRIDGE_PASSWORD =
+            new SecureTokenGenerator(9, new SecureRandom(), PASSWORD);
+
+    private final Random random;
+    private final char[] characters;
+    public int getTokenLength() {
+        return characters.length;
+    }
+    private final int length;
+
+    private SecureTokenGenerator(int length, Random random, String characters) {
+        checkArgument(length > 1);
+        checkArgument(characters != null && characters.length() >= 2);
+
+        this.length = length;
+        this.random = Objects.requireNonNull(random);
+        this.characters = characters.toCharArray();
+    }
+
+    /**
+     * Create session identifiers. This is 4.36e+37 unique values,
+     * which is enough for a good sessions key.
+     */
+    private SecureTokenGenerator() {
+        this(21, new SecureRandom(), ALPHANUMERIC);
+    }
+
+    public String nextToken() {
+        final char[] buffer = new char[length];
+        for (int i = 0; i < buffer.length; ++i) {
+            buffer[i] = characters[random.nextInt(characters.length)];
+        }
+        return new String(buffer);
+    }
+
+    /**
+     * Guaranteed to generate a random password string compatible with Bridge
+     * @return a random password string compatible with Bridge
+     */
+    public String nextBridgePassword() {
+        String token = nextToken();
+        while(!isValidBridgePassword(token)) {
+            token = nextToken();
+        }
+        return token;
+    }
+
+    public boolean isValidBridgePassword(String password) {
+        boolean containsUppercase = false;
+        boolean containsLowercase = false;
+        boolean containsNumeric = false;
+        boolean containsSpecial = false;
+
+        for (int i = 0; i < password.length(); i++) {
+            String character = Character.toString(password.charAt(i));
+            containsUppercase |= UPPERCASE_ALPHA.contains(character);
```

Comment:
This has been replaced by Alx's PasswordGenerator.

However, it is a good catch. I think this worked because Java booleans are stored as just bit 1 or bit 0, but you are right that it is the bitwise OR and not logical OR.